AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -417,9 +422,8 @@ - (void)_reloadDataWithAnimationOptions:(ASDataControllerAnimationOptions)animat
 
     [self accessDataSourceSynchronously:synchronously withBlock:^{
       NSUInteger sectionCount = [_dataSource numberOfSectionsInDataController:self];
-      NSMutableArray *updatedNodes = [NSMutableArray array];
-      NSMutableArray *updatedIndexPaths = [NSMutableArray array];
-      [self _populateFromEntireDataSourceWithMutableNodes:updatedNodes mutableIndexPaths:updatedIndexPaths];
+      NSMutableArray<ASIndexedNodeContext *> *contexts = [NSMutableArray array];
```

Comment:
As a future optimization, we should probably have the method below return the array, so that internally it can create it with a capacity according to the delegate's returned counts.  This may not be practical though, not sure.